src/qiskit_iqm/fake_backends/iqm_fake_backend.py

```diff
@@ -0,0 +1,180 @@
+                    self.chip_sample.two_qubit_gate_durations[gate],
+                ).tensor(
+                    thermal_relaxation_error(
+                        self.chip_sample.t1s[second_qubit],
+                        self.chip_sample.t2s[second_qubit],
+                        self.chip_sample.two_qubit_gate_durations[gate],
+                    )
+                )
+                depolarizing_channel = depolarizing_error(
+                    self.chip_sample.two_qubit_gate_depolarization_rates[gate][connection], 2
+                )
+                full_error_channel = thermal_relaxation_channel.compose(depolarizing_channel)
+                noise_model.add_quantum_error(full_error_channel, gate, [first_qubit, second_qubit])
+
+        return noise_model
+
+    def run(self, run_input: Union[QuantumCircuit, list[QuantumCircuit]], **options) -> JobV1:
+        """
+        Run `run_input` on the fake backend using a simulator.
+
+        This method runs circuit jobs (an individual or a list of QuantumCircuit
+        ) and returns a :class:`~qiskit.providers.JobV1` object.
+
+        It will run the simulation with a noise model of the fake backend (e.g. Adonis).
+
+        Args:
+            run_input (QuantumCircuit or list of QuantumCircuit): An
+                individual or a list of
+                :class:`~qiskit.circuits.QuantumCircuit` objects to run on the backend.
+            options: Any kwarg options to pass to the backend.
+        Returns:
+            The job object representing the run.
+        Raises:
+            ValueError: If empty list of circuits is provided.
+        """
+        circuits = [run_input] if isinstance(run_input, QuantumCircuit) else run_input
+
+        if len(circuits) == 0:
+            raise ValueError("Empty list of circuits submitted for execution.")
+
+        shots = options.get("shots", self.options.shots)
+
+        # COUPLING MAP
+        coupling_map = CouplingMap(self.qubit_connectivity)
+        # otherwise it would throw an error in the transpiler that Flipping gate direction is not supported for CPhase
+        coupling_map.make_symmetric()
+
+        # Create noisy simulator backend and run circuits
+        sim_noise = AerSimulator(noise_model=self.noise_model, coupling_map=coupling_map, basis_gates=self.basis_gates)
+        job = sim_noise.run(circuits, shots=shots)
+
+        return job
+
+
+adonis_chip_sample = IQMChipSample(
+    quantum_architecture=Adonis(),
+    t1s={0: 27000.0, 1: 33000.0, 2: 25000.0, 3: 40000.0, 4: 25000.0},
+    t2s={0: 20000.0, 1: 26000.0, 2: 23000.0, 3: 26000.0, 4: 7000.0},
+    one_qubit_gate_depolarization_rates={"r": {0: 0.0006, 1: 0.0054, 2: 0.0001, 3: 0.0, 4: 0.0005}},
+    two_qubit_gate_depolarization_rates={"cz": {(0, 2): 0.0335, (1, 2): 0.0344, (3, 2): 0.0192, (4, 2): 0.0373}},
+    one_qubit_gate_durations={"r": 40.0},
+    two_qubit_gate_durations={"cz": 80.0},
+    id_="sample-chip",
+)
+
+
+class IQMFakeAdonis(IQMFakeBackend):
```

Comment:
Suggesting to move `IQMFakeAdonis` to its own file, which would be in line with how fake backends are defined in Qiskit. Also provides a nicer structure for the code IMO.

Reply:
...along with `adonis_chip_sample` below

Reply:
Created separate file `iqm_fake_adonis.py`.